util/containerdutil/manifest.go

```diff
@@ -109,3 +116,43 @@ func ManifestDesc(ctx context.Context, provider content.Provider, image ocispec.
 	}
 	return m[0], nil
 }
+
+// Forked from github.com/containerd/containerd/images/image.go
+
+// unknownDocument represents a manifest, manifest list, or index that has not
+// yet been validated.
+type unknownDocument struct {
+	MediaType string          `json:"mediaType,omitempty"`
+	Config    json.RawMessage `json:"config,omitempty"`
+	Layers    json.RawMessage `json:"layers,omitempty"`
+	Manifests json.RawMessage `json:"manifests,omitempty"`
+	FSLayers  json.RawMessage `json:"fsLayers,omitempty"` // schema 1
+}
+
+// ValidateMediaType returns an error if the byte slice is invalid JSON or if
+// the media type identifies the blob as one format but it contains elements of
+// another format.
+func ValidateMediaType(b []byte, mt string) error {
```

Comment:
Can we just update containerd to expose https://github.com/containerd/containerd/blob/a776a27af54a803657d002e7574a4425b3949f56/images/image.go#L401 as a public function?

Can be a follow-up PR.

Reply:
Opened PR https://github.com/containerd/containerd/pull/6266

Reply:
> Can be a follow-up PR.

:+1: to this, though it'd be worth opening an issue to track it.